src/action/file_stat.rs

```diff
@@ -0,0 +1,532 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! A handler and associated types for the file stat action.
+//!
+//! A file stat action responses with stat of a given file
+
+use std::fs::{self, File, Metadata};
+use std::path::{Path, PathBuf};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use log::warn;
+use rrg_proto::{GetFileStatRequest, path_spec::Options, path_spec::PathType,
+                stat_entry::ExtAttr, StatEntry};
+
+use crate::session::{self, Error, Session};
+
+impl From<std::io::Error> for Error {
+    fn from(e: std::io::Error) -> Error {
+        Error::action(e)
+    }
+}
+
+#[derive(Debug)]
+pub struct Response {
+    mode: Option<u64>,
+    inode: Option<u32>,
+    device: Option<u32>,
+    hard_links: Option<u32>,
+    uid: Option<u32>,
+    gid: Option<u32>,
+    size: Option<u64>,
+    access_time: Option<SystemTime>,
+    modification_time: Option<SystemTime>,
+    status_change_time: Option<SystemTime>,
+    blocks_number: Option<u32>,
+    block_size: Option<u32>,
+    represented_device: Option<u32>,
+    flags_linux: Option<u32>,
+    symlink: Option<PathBuf>,
+    pathspec: PathSpec,
+    extended_attributes: Vec<ExtAttr>,
+}
+
+#[derive(Debug)]
+pub struct Request {
+    pathspec: Option<PathSpec>,
+    collect_ext_attrs: Option<bool>,
+    follow_symlink: Option<bool>,
+}
+
+#[derive(Debug)]
+struct PathSpec {
+    nested_path: Option<Box<PathSpec>>,
```

Comment:
`nested_paths` are used only with combination with TSK, no need for this for now.

Reply:
Well, the main reason I had to add this is that without `nested_paths` support I couldn't test this action through the UI. However, I can always remove this feature if needed.